extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java

```diff
@@ -669,6 +672,40 @@ private static List<DecoratorBuildItem> createAnnotationDecorators(Optional<Proj
         return result;
     }
 
+    /**
+     * Create a decorator that sets the port to the http probe.
+     * The rules for setting the probe are the following:
+     * 1. if 'http-action-port' is set, use that.
+     * 2. if 'http-action-port-name' is set, use that to lookup the port value.
+     * 3. if a `KubernetesPorbePortBuild` is set, then use that to lookup the port.
+     * 4. if we still haven't found a port fallback to 8080.
+     *
+     * @return a decorator for configures the port of the http action of the probe.
+     */
+    public static DecoratorBuildItem createProbeHttpPortDecorator(String name, String target, ProbeConfig probeConfig,
+            Optional<KubernetesProbePortNameBuildItem> portName,
```

Comment:
That's awesome! I just have to emit that build item if the management interface is used.

Reply:
That's exactly the idea.